src/set_timeout.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+import logging
+import os
+import sys
+from datetime import datetime, timedelta
+from time import sleep
+
+import kernelci
+import kernelci.config
+import kernelci.db
+from kernelci.cli import Args, Command, parse_opts
+
+from logger import Logger
+
+
+class SetTimeout:
+
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._logger = Logger("config/logger.conf", "set_timeout")
+
+    def _set_timeout_status(self, node):
+        """Set Node status to timeout if maximum wait time is over"""
+        current_time = datetime.utcnow()
+        max_wait_time = datetime.strptime(node['created'],
+                                          '%Y-%m-%dT%H:%M:%S.%f') + \
+            timedelta(hours=node['max_wait_time'])
+        if current_time > max_wait_time:
+            node['status'] = "timeout"
+            self._db.submit({'node': node})
+
+    def run(self):
+        self._logger.log_message(logging.INFO,
+                                 "Setting node status to timeout...")
+        self._logger.log_message(logging.INFO, "Press Ctrl-C to stop.")
+
+        try:
+            while True:
+                nodes = self._db.get_nodes({"status": "pending"})
+                for node in nodes:
+                    if node['parent'] is None:
+                        # Do not set timeout for parent node
+                        continue
```

Comment:
Why is this a special case?

I think however that any pending child nodes would need to be set to timeout too if the parent timeout is reached.

Reply:
I was unsure about when to set a parent's status to timeout. It is possible that a child is created after maximum wait time of the parent. That is why I left it unchanged.
Added `_update_pending_child()` method to set pending child nodes to timeout if the parent is timed out.

Reply:
Yes I think that's fine, if a child gets added later then that's a special case. We could "lock" nodes if the state has to be consistent and not accept child nodes to be added once a parent node is complete, but I don't think it's necessarily going to be a problem.

Reply:
Okay. So, I think we can address that issue later if necessary.